core/modules/layout/plugins/access/entity_id_layout_access.inc

```diff
@@ -28,12 +28,29 @@ class EntityIDLayoutAccess extends LayoutAccessNegatable {
     $form['entity_id'] = array(
       '#title' => t('@entity @entity_id_key is', array('@entity' => $this->entity_info['label'], '@entity_id_key' => strtoupper($this->entity_info['entity keys']['id']))),
       '#type' => 'number',
+      '#required' => TRUE,
       '#min' => 1,
       '#default_value' => is_null($this->settings['entity_id']) ? NULL : (int) $this->settings['entity_id'],
       '#weight' => 100,
     );
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  function formValidate($form, &$form_state) {
+    parent::formValidate($form, $form_state);
+
+    $entity_id = $form_state['values']['entity_id'];
+    if (is_numeric($entity_id) && round($entity_id) == $entity_id && $entity_id >= 1) {
+      $entity = entity_load($this->entity_type, $entity_id);
+      if (!$entity) {
+        $label = $this->entity_info['label'];
+        form_set_error('entity_id', t('There is no @label with ID @entity_id.', array('@label' => $label, '@entity_id' => $entity_id)));
```

Comment:
LGTM, just a tiny coding standard request:

```
         form_set_error('entity_id', t('There is no @label with ID @entity_id.', array(
           '@label' => $label,
           '@entity_id' => $entity_id,
         )));
```
That's only for better readability.

Reply:
Sure, done. Also for the similar constructions in function `form()` and `summary()` in that same file.